src/waypoint.jsx

```diff
@@ -333,13 +352,18 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
```

Comment:
Ah, right, I'd forgotten about that. That'd be quite inconvenient/frustrating. 

Reading through [this thread](https://github.com/facebook/react/issues/4936#issuecomment-179897923) on the topic, it seems that an intermediary HOC may be the better approach. See this [example](https://github.com/ctrlplusb/react-sizeme/blob/master/src/sizeMe.js#L25). 

For example, user would still use 

```jsx
<Waypoint onEnter={this.handleWaypointEnter}>
  <div>
    Something interesting would go here.
  </div>
</Waypoint>
```

and `Waypoint`'s render method would change to something like: 

```jsx
if (this.props.children) {
  return <WaypointRefContainer ref={this.refElement}>{this.props.children}</WaypointRefContainer>;
}
return <span ref={this.refElement} style={{ fontSize: 0 }} />;
```

with `WaypointRefContainer` defined as, eg; 

```jsx
class WaypointRefContainer extends React.Component {
  render() {
    return React.Children.only(this.props.children);
  }
}
```

Thoughts?

(This approach wouldn't allow multi-element `children`, would it?)